app/views/topStories/TopStoriesList.js

```diff
@@ -0,0 +1,60 @@
+
+import React from 'react'
+import {
+	View,
+  ListView,
+	Text,
+	TouchableHighlight,
+} from 'react-native';
+import TopStoriesItem from './TopStoriesItem';
+
+var css = require('../../styles/css');
+
+export default class TopStoriesList extends React.Component {
+
+  constructor(props){
+    super(props);
+
+		this.state = {
+			topStoriesRenderAllRows: false
+		};
+
+    this.datasource = new ListView.DataSource({rowHasChanged: (r1, r2) => r1 !== r2});
+  }
+
+  render() {
+    var topStoriesData = [];
+    if (this.state.topStoriesRenderAllRows){
+      topStoriesData = this.props.data;
+    } else {
+      topStoriesData = this.props.data.slice(0, this.props.defaultResults);
+    }
+
+    var topStoriesDatasource = this.datasource.cloneWithRows(topStoriesData);
+
+    return (
+			<View>
+	      <ListView
+	        dataSource={topStoriesDatasource}
+	        renderRow={ (row) => <TopStoriesItem data={row} navigator={this.props.navigator} /> }
+	        style={css.wf_listview} />
+
+	      {this.state.topStoriesRenderAllRows === false ? (
+	        <TouchableHighlight underlayColor={'rgba(200,200,200,.1)'} onPress={ () => this.setState({topStoriesRenderAllRows: true}) }>
```

Comment:
@jpknoll agreed  @c3bryant @a6wu what do you think, should I create a place like /app/common and create a "button" component in there which can encapsulate what buttons in your app look like (with the touchable highlight, colors, size, etc)?